### PR TITLE
🧪 test: add tests for parse_site_location edge cases

### PR DESCRIPTION
**Pull request by rnovatorov**
🎯 **What:** The testing gap for `parse_site_location` in `src/enapter/cli/http/api/site_location.py` has been addressed. The function parses site location coordinates strings and previously lacked unit test coverage.
📊 **Coverage:** Test coverage has been added for valid locations, locations with extra spaces around names and values, as well as several exceptions: missing values, extra values, non-numeric latitude/longitude coordinates, and empty string handling.
✨ **Result:** Test coverage for `parse_site_location` is fully covered, preventing potential future regressions and making its behavior verifiable and reliable.

---
*PR created automatically by Jules for task [5468704340888702486](https://jules.google.com/task/5468704340888702486) started by @rnovatorov*

### CODE REVIEW

File: tests/unit/test_cli/test_http/test_api/test_site_location.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from enapter.cli.http.api.site_location import parse_site_location
+
+
+def test_parse_site_location_valid():
+    assert parse_site_location("Berlin,52.52,13.405") == ("Berlin", 52.52, 13.405)
+
+
+def test_parse_site_location_with_spaces():
+    # Note: name keeps whitespace, float() handles surrounding whitespace
+    assert parse_site_location(" Berlin , 52.52 , 13.405 ") == (
```
Comment (rnovatorov) [inline tests/unit/test_cli/test_http/test_api/test_site_location.py:14]: @jules, can we strip whitespace?
